packages/google_maps_flutter/lib/src/marker_updates.dart

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of google_maps_flutter;
+
+/// Types of [Marker] updates.
+enum MarkerUpdateEventType {
+  add,
+  update,
+  remove,
+}
+
+Map<MarkerId, Marker> _toMap(Set<Marker> markers) {
+  final Map<MarkerId, Marker> result = <MarkerId, Marker>{};
+  if (markers == null) {
+    return null;
+  }
+  markers.forEach((Marker marker) {
+    if (marker != null) {
+      result[marker.markerId] = marker;
+    }
+  });
+  return result;
+}
+
+/// [Marker] update events to be applied to the [GoogleMap].
+///
+/// Used in [GoogleMapController] when the map is updated.
+@immutable
+class MarkerUpdates {
```

Comment:
If we end up keeping this class we probably don't want it to be public API.

Reply:
Will make this private.

Reply:
still need to do this